oracle_server/oracle_server.go

```diff
@@ -479,8 +476,10 @@ func (os *OracleServer) doReport(curRoundCommitmentHash common.Hash, lastRoundDa
 	auth.GasTipCap = new(big.Int).SetUint64(os.gasTipCap)
 	auth.GasLimit = uint64(3000000)
 
-	// if there is no last round data, then we just submit the curRndCommitHash hash of current round.
-	if lastRoundData == nil {
+	// if there is no last round data or there were missing datapoint in last round data, then we just submit the
+	// commitment hash of current round as data might be available at current round. This vote will be reimbursed by the
+	// protocol, however it won't be abused as it is limited by the 1 vote per round rule.
+	if lastRoundData == nil || lastRoundData.MissingData {
 		var reports []contract.IOracleReport
 		return os.oracleContract.Vote(auth, new(big.Int).SetBytes(curRoundCommitmentHash.Bytes()), reports, types.InvalidSalt, config.Version)
```

Comment:
wouldn't this cause for next round vote also revert. bcause this commitment is for invalid price and we won't allow invalid price anymore. so next round validator is punished.

Reply:
Not at all, because in this context, we just need to vote with the current round commitment hash since only the commitment hash will be stored on-chain. And the vote will return from here: https://github.com/autonity/autonity/blob/develop/autonity/solidity/contracts/Oracle.sol#L167 which mean it stop the processing of the other part of data.